### PR TITLE
Sang hun

### PR DESCRIPTION
**Pull request by sanghoonsim**
Due to the lack of memory, I made some changes for companion/index.js to use local stroage and minor changes for app/index.js.

### CODE REVIEW

**Review by tparanjpe — APPROVED** [review]

File: application/package.json [hunk 1]
```diff
@@ -4,13 +4,15 @@
...
     "appUUID": "aa8f1710-b26f-45f7-83d6-bd9ce1924aea",
-    "appType": "app",
+    "appType": "clockface",
     "appDisplayName": "StressManager",
     "iconFile": "resources/icon.png",
     "wipeColor": "#3f51b5",
     "requestedPermissions": [
       "access_heart_rate",
-      "access_activity"
+      "access_internet",
```
Comment (aderbique) [inline application/package.json:13]: Is this necessary for the device?

File: application/package.json [hunk 2]
```diff
@@ -4,13 +4,15 @@
   },
   "fitbit": {
     "appUUID": "aa8f1710-b26f-45f7-83d6-bd9ce1924aea",
```
Comment (aderbique) [inline application/package.json:6]: Should this be hard coded?

File: application/companion/index.js [hunk 3]
```diff
@@ -62,48 +62,61 @@ async function processFile(endpoint, fileName, data){
...
+      flag = 2;
+    }
+    
+    // To send the message for device to delete just transferred file
+    file_prefix = payload.substring(9,22);
+    // Stroe data in Key-Value store
+    var json_data = JSON.parse(payload);
+    
+    // It implements only when peer connection established
+    if (messaging.peerSocket.readyState === messaging.peerSocket.OPEN) {
```
Comment (aderbique) [inline application/companion/index.js:91]: Do you need to close the socket once transfer is complete?

File: application/companion/index.js [hunk 4]
```diff
@@ -62,48 +62,61 @@ async function processFile(endpoint, fileName, data){
...
+          } else {
+          var existed_data = localStorage.getItem(storage_name);
+          var json_existed = JSON.parse(existed_data);
+          var merged = json_existed.concat(json_data);
+          //console.log("merged: " + JSON.stringify(merged));
+          localStorage.setItem(storage_name, JSON.stringify(merged));
+          }
+          var data_upload = localStorage.getItem(storage_name);
+            
+          var file_prefix_tr = data_upload.substring(9,22);
```
Comment (aderbique) [inline application/companion/index.js:105]: What is the 9 and 22 about?

File: application/companion/index.js [hunk 5]
```diff
@@ -62,48 +62,61 @@ async function processFile(endpoint, fileName, data){
...
+          //console.log(file_prefix_tr);
+          messaging.peerSocket.send(flag+file_prefix);
+     }
+     
+     var filename = "uniqueId_" + file_prefix_tr + "_" + storage_name;
+     try {
+        processFile(endpoint, filename, localStorage.getItem(storage_name));
```
Comment (aderbique) [inline application/companion/index.js:112]: At some point, this code should be refactored to be more modular.

File: application/companion/index.js [hunk 6]
```diff
@@ -1,18 +1,18 @@
...
+
 //get the current date
 var now = new Date();
 
 //get the first timestamp of data to utilize it as prefix of filename
 var file_prefix;
-
 // API Gateway Endpoint for obtaining signed S3 URLs
-// var endpoint = "https://7ntupb3zc7.execute-api.us-west-2.amazonaws.com";
+var endpoint = "https://7ntupb3zc7.execute-api.us-west-2.amazonaws.com";
```
Comment (aderbique) [inline application/companion/index.js:15]: We should set a configuration file or look at having this endpoint be a configuration option on the companion.

File: application/app/index.js [hunk 7]
```diff
@@ -97,11 +80,14 @@ if (HeartRateSensor) {
     fs.writeSync(fd, encode(json_data1));
     //fs.writeSync(fd, convertToBuffer('\n'));
     fs.closeSync(fd);
-    
-    if (hrm.heartRate > 150) {
+    console.log("Memory used: " + memory.js.used + "/" + memory.js.total);
+    if (hrm.heartRate > 135) {
        // stress level function implements if 30 seconds passed from previous execution or if it is first time to run
        if (timestamp >= temp_interval || temp_interval == 0) {
+          vibration.start("ring");
```
Comment (aderbique) [inline application/app/index.js:87]: This ring should be inside of the showButton() function. Possibly in a try catch block so if there's an error, the ring stops

**Review by aderbique — CHANGES_REQUESTED** [review]

File: application/package.json (hunk 1 above)
Comment (sanghoonsim) [reply application/package.json:13]: I think so. After I did that companion started transferring files to the Internet

File: application/package.json (hunk 2 above)
Comment (sanghoonsim) [reply application/package.json:6]: In Fitbit Studio, this is shown as UI. I just changed the app to be on clockface and companion to reach the Internet.

File: application/companion/index.js (hunk 4 above)
Comment (sanghoonsim) [reply application/companion/index.js:105]: In order to identify when the data is collected, the first timestamp of data should be good option for filename. So 9, 22 is index of time value.

File: application/companion/index.js (hunk 5 above)
Comment (sanghoonsim) [reply application/companion/index.js:112]: Yes, I will do that after this update

File: application/companion/index.js (hunk 3 above)
Comment (sanghoonsim) [reply application/companion/index.js:91]: Yes. I will put the closing the socket after reconstructing code.

File: application/companion/index.js (hunk 6 above)
Comment (sanghoonsim) [reply application/companion/index.js:15]: I will figure out after reconstructing the code.

**Review by aderbique — APPROVED** [review]

**Pull request merged**